astropy/units/format/base.py

```diff
@@ -256,15 +279,34 @@ def _did_you_mean_units(cls, unit: str) -> str:
         return did_you_mean(unit, cls._units, fix=cls._fix_deprecated)
 
     @classmethod
-    def _validate_unit(cls, s: str) -> UnitBase:
-        if s in cls._deprecated_units:
-            alternative = (
-                unit.represents if isinstance(unit := cls._units[s], Unit) else None
-            )
+    def _validate_unit(
+        cls, s: str, deprecations: DeprecatedUnitAction = DeprecatedUnitAction.WARN
+    ) -> UnitBase:
+        if (
```

Comment:
I think the `match` is working out!  But I'd still prefer to forego performance here and just start the `match` after the existing
```
if s in cls._deprecated_units:
    msg = "..."
    if (alternative := ...):
        msg += ...
    match DeprecatedUnitAction(deprecations):
        case DeprecatedUnitAction.CONVERT:
            if alternative:
                return alternative
            warnings.warn(msg + ...)
        case ...
```
It keeps the code easy to follow, with all logic to the various deprecated unit possibilities in one place; performance really is not a worry here.